File: internal/database/postgres/postgres_schema.go
package postgres

type User struct {
	ID         string `gorm:"type:text;primaryKey" json:"id"`
	Email      string `gorm:"type:text;not null" json:"email"`
	Name       string `gorm:"type:text;not null" json:"name"`
	GivenName  string `gorm:"type:text;not null" json:"givenName"`
	FamilyName string `gorm:"type:text;not null" json:"familyName"`
	Picture    string `gorm:"type:text" json:"picture"`
	Locale     string `gorm:"type:text" json:"locale"`
}

type Account struct {
	UserID        string `gorm:"type:text;not null" json:"userID"`
	InstitutionID string `gorm:"type:text;not null" json:"institutionID"`
	AccountID     string `gorm:"type:text;not null" json:"accountID"`
	AccountName   string `gorm:"type:text;not null" json:"accountName"`
	SubType       string `gorm:"type:text;not null" json:"subType"`
	Type          string `gorm:"type:text;not null" json:"type"`
}

type Institution struct {
	ID     string `gorm:"type:text;not null" json:"id"`
	UserID string `gorm:"type:text;not null" json:"userID"`
	Name   string `gorm:"type:text;not null" json:"name"`
}

File: database/postgres/postgres_schema.py
from sqlmodel import Field, SQLModel

class User(SQLModel, table=True):
    __tablename__ = "users" #type: ignore
    id: str = Field(primary_key=True)
    email: str
    name: str
    given_name: str
    family_name: str
    picture: str
    locale: str

class Account(SQLModel, table=True):
    __tablename__ = "accounts" #type: ignore
    id: str = Field(primary_key=True)
    user_id: str
    institution_id: str
    name: str
    sub_type: str
    type: str

class Institution(SQLModel, table=True):
    __tablename__ = "institutions" #type: ignore
    id: str = Field(primary_key=True)
    user_id: str
    name: str
